pkgs/by-name/di/distroshelf/package.nix

```
@@ -0,0 +1,57 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  meson,
+  ninja,
+  wrapGAppsHook4,
+  desktop-file-utils,
+  rustPlatform,
+  cargo,
+  rustc,
+  pkg-config,
+  glib,
+  libadwaita,
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "distroshelf";
+  version = "1.0.6";
+
+  src = fetchFromGitHub {
+    owner = "ranfdev";
+    repo = "DistroShelf";
+    tag = "v${finalAttrs.version}";
+    hash = "sha256-g+NHzz91DcdQE6KVr80ypt+IBz6w3Md27tocnKsm9b0=";
+  };
+
+  cargoDeps = rustPlatform.fetchCargoVendor {
+    inherit (finalAttrs) src;
```

```suggestion
    inherit (finalAttrs) pname version src;
```

To better catch hash mismatch in updates